underscore.js

```diff
@@ -1436,14 +1451,25 @@
   _.escape = createEscaper(escapeMap);
   _.unescape = createEscaper(unescapeMap);
 
-  // If the value of the named `property` is a function then invoke it with the
-  // `object` as context; otherwise, return it.
-  _.result = function(object, prop, fallback) {
-    var value = object == null ? void 0 : object[prop];
-    if (value === void 0) {
-      value = fallback;
+  // If the value at the named `path` is a function then invoke it with its
+  // parent object as context; otherwise, return it.
+  _.result = function(object, path, fallback) {
+    if (!isArray(path)) path = [path];
+    var length = path.length;
+    // If path is `[]`, step through the loop once so `fallback` gets used.
+    if (!length) {
+      length = 1;
+      object = void 0;
+    }
+    for (var i = 0; i < length; i++) {
+      var prop = object == null ? void 0 : object[path[i]];
+      if (prop === void 0) {
+        prop = fallback;
+        i = length; // Ensure we don't continue iterating.
+      }
+      object = _.isFunction(prop) ? prop.call(object) : prop;
     }
-    return _.isFunction(value) ? value.call(object) : value;
+    return object;
```

Comment:
Credit to @jdalton on for this one, as it's lifted almost directly from lodash. I spent a few days picking at it, and couldn't come up with a better implementation.